File: frontend/src/types/area.ts
export interface Area {
  id: number;
  name: string;
  description?: string;
  region?: string;
  display_order: number;
  location_count: number;
  created_at: string;
  updated_at: string;
}

export interface AreaWithLocations {
  area: Area;
  locations: Location[];
}

File: backend/internal/models/area.go
package models

import "time"

// Area represents a geographic grouping for climbing locations
type Area struct {
	ID           int       `json:"id" db:"id"`
	Name         string    `json:"name" db:"name"`
	Description  *string   `json:"description,omitempty" db:"description"`
	Region       *string   `json:"region,omitempty" db:"region"`
	DisplayOrder int       `json:"display_order" db:"display_order"`
	IsActive     bool      `json:"is_active" db:"is_active"`
	CreatedAt    time.Time `json:"created_at" db:"created_at"`
	UpdatedAt    time.Time `json:"updated_at" db:"updated_at"`
}

// AreaWithLocationCount extends Area with location count
type AreaWithLocationCount struct {
	Area
	LocationCount int `json:"location_count" db:"location_count"`
}
